### PR TITLE
NetStandard compatibility changes

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
I've backported some changes from netcore branch to the dev branch

### CODE REVIEW

File: src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-//     Copyright (C) 2013-2015 Akka.NET project <https://github.com/akkadotnet/akka.net>
-// </copyright>
-//-----------------------------------------------------------------------
-
-using Akka.Configuration;
-using Akka.Dispatch;
-
-namespace Akka.Tests.Performance.Dispatch
-{
-    public class ThreadPoolDispatcherColdThroughputSpec : ColdDispatcherThroughputSpecBase
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs]: Why was this deleted?

Comment (alexvaluyskiy) [inline src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs]: We should or delete it, or add to the project (csproj)

**Comment by alexvaluyskiy** [comment]
I've got this error in ConvergenceSpecTests
```
[ERROR][8/27/2016 6:41:33 PM][Thread 0019][akka://MultiNodeClusterSpec/system/log1-DefaultLogger] Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
Cause: System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at Akka.Event.DefaultLogMessageFormatter.Format(String format, Object[] args) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\DefaultLogMessageFormatter.cs:line 23
   at Akka.Event.LogMessage.ToString() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\LogMessage.cs:line 42
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at Akka.Event.LogEvent.ToString() in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\LogEvent.cs:line 65
   at Akka.Event.StandardOutLogger.PrintLogEvent(LogEvent logEvent) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\StandardOutLogger.cs:line 132
   at Akka.Event.DefaultLogger.Print(LogEvent logEvent) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\DefaultLogger.cs:line 39
   at Akka.Event.DefaultLogger.Receive(Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Event\DefaultLogger.cs:line 29
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorBase.cs:line 154
   at Akka.Actor.ActorCell.ReceiveMessage(Object message) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 168
   at Akka.Actor.ActorCell.Invoke(Envelope envelope) in D:\BuildAgent\work\d26c9d7f36545acd\src\core\Akka\Actor\ActorCell.DefaultMessages.cs:line 77
```

**Comment by Horusiath** [comment]
@alexvaluyskiy this stack trace suggest problem with message format somewhere in the logging - something like `Log.Info("{0} {1}, parameter0)`.

**Comment by alexvaluyskiy** [comment]
@Horusiath I know, but I don't have such changes

File: src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs (hunk 1 above)
Comment (alexvaluyskiy) [inline src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs]: I've removed it, because we have the same classes in DefaultDispatcherColdThroughputSpec.cs file

File: src/core/Akka.Streams.Tests/ActorMaterializerSpec.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 namespace Akka.Streams.Tests
 {
-    public class ActorMaterializerSpec : AkkaSpec
+    public class ActorMaterializerSpec : TestKit.Tests.AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Streams.Tests/ActorMaterializerSpec.cs]: I did it to avoid conflicts between AkkaSpec in Streams and AkkaSpec in TestKit

File: src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ protected bool Receiver(object message)
             else if (message is DeleteMessagesSuccess)
             {
                 if (AskedForDelete == null)
-                    throw new ApplicationException("Received DeleteMessagesSuccess without anyone asking for delete!");
+                    throw new ArgumentNullException("Received DeleteMessagesSuccess without anyone asking for delete!");
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs]: ApplicationException is absent in netcore

File: src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ protected bool Receiver(object message)
             else if (message is DeleteMessagesSuccess)
             {
                 if (AskedForDelete == null)
-                    throw new ApplicationException("Received DeleteMessagesSuccess without anyone asking for delete!");
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/PersistentActorSpec.Actors.cs]: ApplicationException is absent in netcore

File: src/core/Akka.Streams.TestKit.Tests/TestException.cs [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 namespace Akka.Streams.TestKit.Tests
 {
-    public class TestException : SystemException
```
Comment (alexvaluyskiy) [inline src/core/Akka.Streams.TestKit.Tests/TestException.cs]: SystemException is absent in netcore

File: src/core/Akka.Streams/Dsl/Tcp.cs [hunk 6]
```diff
@@ -118,7 +118,7 @@ public TcpExt(ExtendedActorSystem system)
             IImmutableList<Inet.SocketOption> options = null, bool halfClose = false, TimeSpan? idleTimeout = null)
         {
             // DnsEnpoint isn't allowed
-            var ipAddresses = System.Net.Dns.GetHostAddresses(host);
+            var ipAddresses = System.Net.Dns.GetHostAddressesAsync(host).Result;
```
Comment (alexvaluyskiy) [inline src/core/Akka.Streams/Dsl/Tcp.cs]: Sync methods are absent in netcore

File: src/core/Akka.Tests/Actor/ActorPathSpec.cs [hunk 7]
```diff
@@ -226,7 +226,7 @@ public void Validate_element_parts(string element, bool matches)
         [InlineData("Using parenthesis(4711)")]
         public void Validate_that_url_encoded_values_are_valid_element_parts(string element)
         {
-            var urlEncode = WebUtility.UrlEncode(element);
+            var urlEncode = System.Net.WebUtility.UrlEncode(element);
```
Comment (alexvaluyskiy) [inline src/core/Akka.Tests/Actor/ActorPathSpec.cs]: There is no WebUtility in netcore

File: src/core/Akka/Actor/ActorSelection.cs [hunk 8]
```diff
@@ -348,7 +348,7 @@ public SelectChildPattern(string patternStr)
         /// Returns a <see cref="string" /> that represents this instance.
         /// </summary>
         /// <returns>A <see cref="string" /> that represents this instance.</returns>
-        public override string ToString() => PatternStr.ToString(CultureInfo.InvariantCulture);
+        public override string ToString() => PatternStr.ToString();
```
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/ActorSelection.cs]: This overloaded method is not supported in netcore, and don't have any sense in .NET 45
http://referencesource.microsoft.com/#mscorlib/system/string.cs,2648

File: src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs (hunk 1 above)
Comment (Aaronontheweb) [inline src/core/Akka.Tests.Performance/Dispatch/ThreadPoolDispatcherThroughputSpec.cs]: Ah, I see.
